CMakeLists.txt

```diff
@@ -54,24 +53,30 @@ add_library(graphqlservice
   GraphQLResponse.cpp
   GraphQLService.cpp
   Introspection.cpp
-  ${CMAKE_BINARY_DIR}/graphqlservice/IntrospectionSchema.cpp)
+  IntrospectionSchema.cpp)
```

Comment:
Is it correct to remove the `${CMAKE_BINARY_DIR}/` part?

In out-of-source build, I believe `schemagen` will run in the CMAKE_CURRENT_BINARY_DIR (CMAKE_BINARY_DIR also worked since there is only one root CMakeLists.txt).
Relative outputs of the of the command will thus be in this build tree.
`add_library` sources are relative to the current source directory (CMAKE_CURRENT_SOURCE_DIR), if the files are rooted somewhere else, e.g. the build tree, we have to use a more explicit path.

I believe it should be `${CMAKE_CURRENT_BINARY_DIR}/IntrospectionSchema.cpp`.

Reply:
In this case, I think it is correct. The `add_custom_command` command creates targets for the `OUTPUT` names, so by using the same name as that value in `add_custom_command` I think we're actually referencing the target rather than a specific file path.

Reply:
Indeed, it works even for out of source builds.

Reading `target_sources` documentation, which is linked by `add_library`, and seems to be more descriptive of this `source` arguments, I don't see anything about possibly being targets:

> Specify sources to use when compiling a given target. Relative source file paths are interpreted as being relative to the current source directory (i.e. CMAKE_CURRENT_SOURCE_DIR).
> -- https://cmake.org/cmake/help/v3.13/command/target_sources.html#command:target_sources

But I found a CMake test, which seems to indicated this is a tested use case:
- https://github.com/Kitware/CMake/blob/07cfb18f9d29cfc0588ede928846a03ec5599c48/Tests/CustomCommand/CMakeLists.txt#L188

I'm a bit uneasy that it seems undocumented but it seems unlikely they would break this, as it is explicitly tested.